core/src/main/java/com/linecorp/armeria/common/stream/AbstractStreamMessageDuplicator.java

```diff
@@ -657,4 +744,137 @@ public String toString() {
             }
         }
     }
+
+    /**
+     * A CircularQueue that stores signals in order and retrieves by {@link #get(int)}.
+     * Adding and removing elements are happened by only one thread, or at least once at a time. Reading
```

Comment:
Adding and removing elements are happened by only one thread -> Addition and removal of elements are done by only one thread

happened -> done ('happen' implies 'unexpectedly')